scripts/pylib/twister/twisterlib/coverage.py

```diff
@@ -340,7 +340,7 @@ def run_coverage(testplan, options):
 
     logger.info("Generating coverage files...")
     coverage_tool = CoverageTool.factory(options.coverage_tool)
-    coverage_tool.gcov_tool = options.gcov_tool
+    coverage_tool.gcov_tool = str(options.gcov_tool)
```

Comment:
why `options.gcov_tool` is empty in your case, with what command line arguments and env variables ?

Reply:
See https://github.com/zephyrproject-rtos/zephyr/issues/66897, `./scripts/twister -T tests/arch/x86/nmi/ --coverage`

Reply:
> See #66897, `./scripts/twister -T tests/arch/x86/nmi/ --coverage`

Can't reproduce with it. Is `ZEPHYR_SDK_INSTALL_DIR` or `ZEPHYR_TOOLCHAIN_VARIANT` set ?

Reply:
Mind you that's on a machine with only a couple of toolchains installed, on current main it fails gracefully with

`ERROR   - Can't find a suitable gcov tool. Use --gcov-tool or set ZEPHYR_SDK_INSTALL_DIR.`

Reply:
This PR seems to deal with a symptom of misconfigured environment, and with empty gcov path twister either makes empty coverage report, or just uses some arbitrary gcov found on the host.

> `ERROR - Can't find a suitable gcov tool. Use --gcov-tool or set ZEPHYR_SDK_INSTALL_DIR.`

it is from the new fix just merged
https://github.com/zephyrproject-rtos/zephyr/pull/66747/files#diff-d7590c92e90c3827d0cac169343ceaac1ba22223aec878f5ccef89ea3e035b1aR355